### DISCUSSION TITLE
How should empty tags be treated?

### DISCUSSION THREAD

**Discussion by agentgt** [mustache/spec#151] [Q&A]
Consider the case:

```
{{}}hello{{}}
```

I found a lot of varying behavior in mustache implementations.

* In the wontache playground you get an empty string. 
* In Mustache.js you get `hello`. The empty tags appear to be ignored.
* JStachio fails on the second empty tag because it can't handle two empty tags in a row.
* JMustache fails horribly with index out of bounds
* Mustache.java fails with an Empty mustache exception
* Handlebars.js (albeit not mustache) fails on the first empty tag

Perhaps it is out of scope for the spec to dictate what happens on empty tags? That being said I do think Wontache's behavior is incorrect regardless.

**Comment by jgonggrijp** [comment]
Wontache produces the empty string because it interprets the whole template as a single interpolation tag with `}}hello{{` as the key. This playground savestate demonstrates it:

```js
{"data":{"text":"{\n    '}}hello{{': 'x'\n}"},"templates":[{"name":"","text":"{{}}hello{{}}"}]}
```

Not meaning to defend the behavior though, just clarifying.

I see no use in empty tags, so I don't think it needs specifying, but I'll leave this open for a while so different opinions can be heard.

**Comment by determin1st** [comment]
mine leaves them alone,
both the empty and blank names

`{{}}hello{{}}` => `{{}}hello{{}}`
`{{   }}hello{{}}` => `{{   }}hello{{}}`

**Comment by determin1st** [comment]
oke, now i think that there should be a parse exception - empty clause have no use
